apps/frontend/src/components/timeline/message-event.tsx

```diff
@@ -18,9 +20,21 @@ interface MessageEventProps {
 export function MessageEvent({ event }: MessageEventProps) {
```

Comment:
I feel like it might be time to split this component into sub components now? I foresee more variance here, and it's already kinda messy, so let's try out a FailedMessageEvent, PendingMessageEvent, and the normal message event, all wrapped in this one MessageEvent if that makes sense?

Reply:
Done! Split into:

- `MessageLayout` - shared structure (avatar, content wrapper, status slot, actions slot)
- `SentMessageEvent` - normal sent messages with timestamp
- `PendingMessageEvent` - reduced opacity, delayed "Sending..." indicator
- `FailedMessageEvent` - red border, "Failed to send", retry button

The main `MessageEvent` now just dispatches based on status via a switch statement. Much cleaner for future variance!